lib/active_reporting/report.rb

```diff
@@ -84,8 +85,10 @@ def select_aggregate
       end
     end
 
-    def dimension_joins
-      @dimensions.select { |d| d.type == :standard }.map { |d| d.name.to_sym }
+    def dimension_joins(join_method = ReportingDimension::JOIN_METHODS[:joins])
+      @dimensions.select { |d| d.type == Dimension::TYPES[:standard] }.
+                  select { |d| d.join_method == join_method }.
```

Comment:
We should be able to combine the two `select` calls here, right? Is there a reason why you're wanting to loop twice?

```ruby
select { |d| d.type == Dimension::TYPES[:standard] && d.join_method == join_method }
```

Reply:
Done